resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -375,9 +375,14 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
 
       logDebug(s"Received ${offers.size} resource offers.")
 
+      val blacklist = scheduler.nodeBlacklist()
       val (matchedOffers, unmatchedOffers) = offers.asScala.partition { offer =>
-        val offerAttributes = toAttributeMap(offer.getAttributesList)
-        matchesAttributeRequirements(slaveOfferConstraints, offerAttributes)
+        if (blacklist.contains(offer.getHostname)) {
+          false
```

Comment:
This will save many computation on the invalid offers.

Reply:
I don't mind to add this optimisation, but imo, you can't remove original check. Since they happening in two different places and the code that matches offer is "repeated" at lauchTask

Reply:
It seems that you are saying `unmatchedOffers` will be evaluated at `canLaunchTask. Does it what you mean really? At line 392, we simple do `declineUnmatchedOffers(d, unmatchedOffers)`.

Reply:
I mean that even though the offer is matched, then there is code in canLaunchTask that verifies that task can be launched with respect to cpu/memory requirements(so also blacklisting)

Reply:
I said the computation on the invalid offers (not matched).
> This will save many computation on the invalid offers.

Reply:
I agree with @dongjoon-hyun , seems this would be more efficient. I don't understand the objection